File: iOSQRcode/iOSQRcode/SecondViewController.m
//
//  SecondViewController.m
//  iOSQRcode
//
//  Created by ymq on 16/3/21.
//  Copyright © 2016年 ymq. All rights reserved.
//

#import "SecondViewController.h"
#import "QRcodeView.h"

@interface SecondViewController ()

@property (nonatomic, strong) QRcodeView *qrview;
@property (nonatomic, strong) UIActivityIndicatorView *indicatorView;

@end

@implementation SecondViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.view.backgroundColor = [UIColor blackColor];
    [self.navigationController setNavigationBarHidden:YES animated:YES];
    
    _qrview = [[QRcodeView alloc] initWithFrame:CGRectMake(0, 0, self.view.frame.size.width, self.view.frame.size.height)];
    [_qrview setScanResult:^(QRScanResultType type, NSString *result) {
        if (type == QRScanResultType_Succeed)
        {
            NSLog(@"%@",result);
        }
        else
        {
            NSLog(@"扫描失败");
        }
    }];
    [self.view addSubview:_qrview];
    
    _indicatorView = [[UIActivityIndicatorView alloc] init];
    _indicatorView.bounds = CGRectMake(0, 0, 40, 40);
    _indicatorView.center = self.view.center;
    _indicatorView.activityIndicatorViewStyle = UIActivityIndicatorViewStyleWhiteLarge;
    _indicatorView.hidesWhenStopped = YES;
    [self.view addSubview:_indicatorView];
    [_indicatorView startAnimating];
    
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    [_qrview startReading];
    [_indicatorView stopAnimating];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (void)dealloc
{
    NSLog(@"asdfasfas");
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift_TableView/Swift_TableView/SecondViewController.swift
//
//  SecondViewController.swift
//  Swift_TableView
//
//  Created by ymq on 16/3/10.
//  Copyright © 2016年 ymq. All rights reserved.
//

import UIKit

//定义一个闭包
typealias callBack = (_ string:String)->Void


class SecondViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
    @objc var funObj: callBack?
    var count : Int = 0
    
    @objc var _tableView :UITableView!
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "2"
        self.view.backgroundColor = UIColor.red
        
//        let btn = UIButton(type: UIButtonType.Custom)
//        btn.frame = CGRectMake(20, 100, 100, 20)
//        btn.backgroundColor = UIColor.redColor();
//        btn.addTarget(self, action:#selector(self.btnClick), forControlEvents: UIControlEvents.TouchUpInside)
//        self.view.addSubview(btn)
        // Do any additional setup after loading the view.
        _tableView = UITableView(frame: CGRect(x: 0, y: 64, width: ScreenWith, height: ScreenHeight), style: UITableViewStyle.plain)
        
        _tableView.delegate = self
        _tableView.dataSource = self
        _tableView.register(CustomCell.classForCoder(), forCellReuseIdentifier: "CELLID")
        self.view.addSubview(_tableView!);
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @objc func btnClick()
    {
        funObj?("111")
        print("click")
    }
    
    internal func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return 10;
    }
    
    internal func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
    {
        let cell = tableView.dequeueReusableCell(withIdentifier: "CELLID") as! CustomCell
        cell.lab.text = "\(indexPath.row)"
    
        cell.click = {
            //[unowned self] 标志，防止闭包循环引用，以致界面释放不掉
             [unowned self](titleStr: NSString) -> Void in
            self.count = 0;
              print("\(titleStr) \(indexPath.row)")
        }
        return cell;
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 300;
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
//        let secondVC = SecondViewController()
        //---1----
//                secondVC.funObj = {
//                    (string:String)->Void in
//                    print("\(string,self.title)")
//                };
        //---2---
//        secondVC.funObj = self.secondCallback
//        self.navigationController?.pushViewController(secondVC, animated: true)
        
        let thirdVC = ThirdViewController()
        self.navigationController?.pushViewController(thirdVC, animated: true)
        
        let cell = tableView.cellForRow(at: indexPath as IndexPath)
        cell?.isSelected = false
        
    }
    
    @objc func secondCallback(string:String) -> Void{
        print("\(string,self.title)")
    }
    

    
    
    //对象释放时调用
    deinit{
        print("界面释放")
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
